river/compose/union.py

```diff
@@ -233,3 +264,36 @@ def transform_one(self, x):
                 *(t.transform_one(x) for t in self.transformers.values())
             )
         )
+
+    # Mini-batch methods
+
+    def learn_many(self, X: pd.DataFrame, y: pd.Series = None):
+        """Update each transformer.
+
+        Parameters
+        ----------
+        x
+            Features.
+        y
+            An optional target, this is expected to be provided if at least one of the transformers
+            is supervised (i.e. it inherits from `base.SupervisedTransformer`).
+
+        """
+        for t in self.transformers.values():
+            if isinstance(t, base.SupervisedTransformer):
+                t.learn_many(X, y)
+            else:
+                t.learn_many(X)
+        return self
+
+    def transform_many(self, X: pd.DataFrame):
+        """Passes the data through each transformer and packs the results together."""
+        # INFO: not the most optimal but at least it works
+        return pd.DataFrame(
+            dict(
+                collections.ChainMap(
+                    *(t.transform_many(X) for t in self.transformers.values())
+                )
+            ),
+            copy=False,
+        )
```

Comment:
Mmm intuitively I would just have used the [`pandas.concat` function](https://pandas.pydata.org/docs/reference/api/pandas.concat.html). What do you think?

Reply:
Sounds reasonable